### PR TITLE
Several fixes affecting `allocate_buffers`

### PR DESCRIPTION
**Pull request by BenBE**

### CODE REVIEW

File: v4l2loopback.c [hunk 1]
```diff
@@ -2187,17 +2187,30 @@ static int allocate_buffers(struct v4l2_loopback_device *dev)
 
 	dprintk("allocating %ld = %ldx%d\n", dev->imagesize, dev->buffer_size,
 		dev->buffers_number);
+	int err = -ENOMEM;
```
Comment (umlaeute) [inline v4l2loopback.c:2190]: ISO C90 forbids mixed declarations and code

**Review by umlaeute — CHANGES_REQUESTED** [review]

File: v4l2loopback.c (hunk 1 above)
Comment (BenBE) [reply v4l2loopback.c]: Fixed.

File: v4l2loopback.c [hunk 2]
```diff
@@ -2552,7 +2552,7 @@ static int v4l2_loopback_add(struct v4l2_loopback_config *conf, int *ret_nr)
 	dprintk("buffer_size = %ld (=%d)\n", dev->buffer_size,
 		dev->pix_format.sizeimage);
 	err = allocate_buffers(dev);
-	if (err && dev->buffer_size)
+	if (err < 0)
```
Comment (umlaeute) [inline v4l2loopback.c:2555]: this fails in the default case (where the image is 0x0, and thus `dev->buffer_size == 0`

File: v4l2loopback.c [hunk 3]
```diff
@@ -2163,6 +2163,12 @@ static int allocate_buffers(struct v4l2_loopback_device *dev)
 	if (0 <= dev->buffer_size)
 		return -EINVAL;
 
+	if (dev->buffer_size < 1 || dev->buffers_number < 1)
```
Comment (umlaeute) [inline v4l2loopback.c:2168]: this partly duplicates the if-clause above

**Comment by umlaeute** [comment]
picked a working commits and fixed the rest.

**Pull request closed without merging**